inst/rmd/qc_report.rmd

```diff
@@ -356,6 +381,41 @@ This can be the result of a low-quality library or may occur if there is no mito
 In such situations, the calculated probability of compromise may not be valid (see [miQC vignette](https://bioconductor.org/packages/3.13/bioc/vignettes/miQC/inst/doc/miQC.html#when-not-to-use-miqc) for more details).
 
 
+```{r}
+if(has_processed){
+  
+  # check that processed SCE contains UMAP and PCA
+  if(!"UMAP" %in% reducedDimNames(processed_sce)){
+    if(!"PCA" %in% reducedDimNames(processed_sce)){
+      
+      # if PCA is missing, check for HVG, if no HVG use all genes 
+      if(!is.null(processed_meta$highly_variable_genes)){
+        subset_genes <- processed_meta$highly_variable_genes
+      }else{
+        subset_genes <- rownames(processed_sce)
+      }
+      processed_sce <- scater::runPCA(processed_sce,
+                                      subset_row = subset_genes)
+    }
+    
+    processed_sce <- scater::runUMAP(processed_sce, 
+                                     dimred = "PCA")
+  }
+  
+  # create UMAP colored by number of genes detected 
+  scater::plotReducedDim(processed_sce, 
+                         dimred = "UMAP", 
```

Comment:
Just simplifying a bit since this is a default `UMAP`
```suggestion
  scater::plotUMAP(processed_sce, 
```

Reply:
Note that this change wasn't just removing `dimred`, but also changing to `plotUMAP()`, which removes the need for that argument (as long as there is a reduced dimension table named UMAP).